pkg/kubectl/cmd/apply.go

```diff
@@ -220,7 +230,13 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *App
 			helper := resource.NewHelper(info.Client, info.Mapping)
 			patcher := NewPatcher(encoder, decoder, info.Mapping, helper, overwrite)
 
-			patchBytes, err := patcher.patch(info.Object, modified, info.Source, info.Namespace, info.Name)
+			infos := []*resource.Info{info}
```

Comment:
If Apply has VersionedObject as nil, other things will fail.  VersionedObject was added in order to solve the Apply use case, so it's expected to work in general.